src/reconcile/rec_write.c

```diff
@@ -668,6 +671,26 @@ __rec_init(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags, WT_SALVAGE_COO
     return (ret);
 }
 
+/*
+ * __rec_clear_datastore_update_list --
+ *     Clear the update cache stored on the reconciliation structure, if called from the error path
+ *     then clear the update datastore flag.
+ */
+static void
+__rec_clear_datastore_update_list(WT_SESSION_IMPL *session, WT_RECONCILE *r, bool error)
```

Comment:
Are you sure we call this before we release the eviction's exclusive lock?

Reply:
Yes this is called before eviction's lock is released, additionally in the error case it is called before the reconciliation page lock is released.